src/internal/analytics-metadata/__tests__/metadata-utils.test.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { GeneratedAnalyticsMetadataFragment } from '../interfaces';
+import { merge, mergeMetadata, processMetadata } from '../metadata-utils';
+
+jest.mock('../labels-utils', () => ({
+  processLabel: (node: HTMLElement | null, label: string) => `processed-${label}`,
+}));
+
+describe('processMetadata', () => {
+  test('recursively identifies elements ending with "label"', () => {
+    expect(processMetadata(null, { label: 'a', entry: { columnLabel: 'b', notLabelEnding: 'c' } })).toEqual({
+      label: 'processed-a',
+      entry: { columnLabel: 'processed-b', notLabelEnding: 'c' },
+    });
+  });
+});
+
+describe('merge', () => {
+  test('returns target when source is null', () => {
```

Comment:
Kinda. It returns a copy of target/source.

Reply:
Yeah, left it as-is